template_workflows/auto_release/release.yaml

```
@@ -8,6 +8,12 @@ on:
 jobs:
   auto-release:
     runs-on: ubuntu-latest
+    # Mainly setting permissions here to appease CodeQL
+    # Actual job permissions come from the dedicated Token we generate in
+    # generate-token
+    permissions:
+      contents: read
```

Since the release operations require write permissions AFAIK, I'd say we keep this top-level `permissions` setting aligned as close as possible (even if we're overriding it) just to help ourselves avoid confusion in the future about required privileges for each workflow.
```suggestion
      contents: write
```